WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/editor/WatchFaceConfigViewModel.kt

```diff
@@ -0,0 +1,282 @@
+
+        val bitmap = editorSession.renderWatchFaceToBitmap(
+            RenderParameters(
+                DrawMode.INTERACTIVE,
+                WatchFaceLayer.ALL_WATCH_FACE_LAYERS,
+                RenderParameters.HighlightLayer(
+                    RenderParameters.HighlightedElement.AllComplicationSlots,
+                    Color.RED, // Red complication highlight.
+                    Color.argb(128, 0, 0, 0) // Darken everything else.
+                )
+            ),
+            editorSession.previewReferenceTimeMillis,
+            previewComplicationData
+        )
+
+        val colorStyle =
+            userStyle[colorStyleKey] as UserStyleSetting.ListUserStyleSetting.ListOption
+        val ticksEnabledStyle =
+            userStyle[drawPipsKey] as UserStyleSetting.BooleanUserStyleSetting.BooleanOption
+        val minuteHandStyle =
+            userStyle[minuteHandLengthKey] as UserStyleSetting.DoubleRangeUserStyleSetting.DoubleRangeOption
+
+        val watchFacePreview = UserStylesAndPreview(
+            colorStyleId = colorStyle.id.toString(),
+            ticksEnabled = ticksEnabledStyle.value,
+            minuteHandLength = multiplyByMultipleForSlider(minuteHandStyle.value).toFloat(),
+            previewImage = bitmap
+        )
+
+        // Triggers listener that's connected to the flow, so the UI gets the updated values.
+        // Uses null as oldValue, as this is only triggered when something changes and there isn't
+        // a reason the keep a variable of it.
+        changeSupport.firePropertyChange(
+            "newWatchFace",
+            null,
+            watchFacePreview)
+    }
+
+    fun setColorStyle(newColorStyle: UserStyleSetting.ListUserStyleSetting.ListOption) {
+        setUserStyleOption(colorStyleKey, newColorStyle)
+    }
+
+    fun setDrawPips(enabled: Boolean) {
+        setUserStyleOption(
+            drawPipsKey,
+            UserStyleSetting.BooleanUserStyleSetting.BooleanOption.from(enabled)
+        )
+    }
+
+    fun setMinuteHandArmLength(newLengthRatio: Float) {
+        val newMinuteHandLengthRatio = newLengthRatio.toDouble() / MULTIPLE_FOR_SLIDER
+
+        setUserStyleOption(
+            minuteHandLengthKey,
+            UserStyleSetting.DoubleRangeUserStyleSetting.DoubleRangeOption(newMinuteHandLengthRatio)
+        )
+    }
+
+    // Saves User Style Option change back to the back to the EditorSession.
+    private fun setUserStyleOption(
+        userStyleSetting: UserStyleSetting,
+        userStyleOption: UserStyleSetting.Option
+    ) {
+        Log.d(WatchFaceConfigActivity.TAG, "setUserStyleOption()")
+        Log.d(WatchFaceConfigActivity.TAG, "\tuserStyleSetting: $userStyleSetting")
+        Log.d(WatchFaceConfigActivity.TAG, "\tuserStyleOption: $userStyleOption")
+
+        // Casts as a [HashMap], so we can update values.
+        val hashmap =
+            userStyle.selectedOptions as HashMap<UserStyleSetting, UserStyleSetting.Option>
```

Comment:
This cast looks unsafe and suspicious, I think there's a `toMutableUserStyle()` to avoid mutating one in-place.

Reply:
Ah, yes, that was just added in 20. I upgraded sample and swapped it.